tpu-provisioner/internal/cloud/gke.go

```diff
@@ -405,6 +405,11 @@ func (g *GKE) nodePoolForPod(name string, p *corev1.Pod) (*containerv1beta1.Node
 		placementPolicy.Type = "COMPACT"
 	}
 
+	var diskType string
```

Comment:
Do we want to do any upfront validation that confidential storage requires hyperdisk, or a disktype =~ "hyperdisk-.*" ?

Should fail fairly quickly when we try to create the node pool as well.

Reply:
No, did want to end up with a hardcoded list of disk types that are out of date with GCP